src/goto-symex/goto_symex.cpp

```diff
@@ -665,3 +669,66 @@ bool goto_symext::constant_propagate_delete_char_at(
 
   return true;
 }
+
+bool goto_symext::constant_propagate_trim(
+  statet &state,
+  symex_assignt &symex_assign,
+  const function_application_exprt &f_l1)
+{
+  const auto &f_type = to_mathematical_function_type(f_l1.function().type());
+  const auto &length_type = f_type.domain().at(0);
+  const auto &char_type = to_pointer_type(f_type.domain().at(1)).subtype();
+
+  const refined_string_exprt &s = to_string_expr(f_l1.arguments().at(2));
+  const auto s_data_opt = try_evaluate_constant_string(state, s.content());
+
+  if(!s_data_opt)
+    return false;
+
+  const array_exprt &string_data = s_data_opt->get();
+  auto &operands = string_data.operands();
+
+  // Note where the trim would start/end.
+  auto start_iter = operands.begin();
+  auto end_iter = operands.end();
+  for(auto it = operands.begin(); it != operands.end(); it++)
+  {
+    auto &constant_value = to_constant_expr(*it);
+    char character = numeric_cast_v<int>(constant_value);
+    if(character > ' ')
+    {
+      start_iter = it;
+      break;
+    }
+  }
+
+  for(auto it = operands.rbegin(); it != operands.rend(); it++)
+  {
+    auto &constant_value = to_constant_expr(*it);
+    char character = numeric_cast_v<int>(constant_value);
+    if(character > ' ')
+    {
+      end_iter = it.base();
+      break;
+    }
+  }
+
+  // Then copy in what the string would be with whitespace removed.
+  exprt::operandst new_operands(start_iter, end_iter);
+  const constant_exprt new_char_array_length =
+    from_integer(new_operands.size(), length_type);
+
+  const array_typet new_char_array_type(char_type, new_char_array_length);
```

Comment:
use curly braces `{}`

Reply:
Where would those go? (And curious about why)

Reply:
`const array_typet new_char_array_type{char_type, new_char_array_length};`
it's the preferred way of constructing an object, here are some reasons why  https://stackoverflow.com/questions/18222926/why-is-list-initialization-using-curly-braces-better-than-the-alternatives